In [1]:
#Loading libraries

import pandas as pd
import numpy as np
import glob

import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn import metrics
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import plot_confusion_matrix

from scipy.stats import ks_2samp

In [2]:
#Loading Datasets

datatran = pd.DataFrame()
colunas = pd.DataFrame()

dir_path = r"C:\Users\Álan Batista\Data\Transito\*.csv"
res = glob.glob(dir_path)

for file in res:
    temp = pd.read_csv(file, encoding="Latin-1", delimiter=";")
    n = len(temp.columns.tolist())
    print(f"Checando por número de colunas: .csv tem {n} colunas.")    
    temp_series = pd.Series(temp.columns.tolist())
    colunas = pd.concat([colunas,temp_series])    
    datatran = pd.concat([datatran,temp])

Checando por número de colunas: .csv tem 26 colunas.
Checando por número de colunas: .csv tem 25 colunas.
Checando por número de colunas: .csv tem 30 colunas.
Checando por número de colunas: .csv tem 30 colunas.
Checando por número de colunas: .csv tem 30 colunas.
Checando por número de colunas: .csv tem 30 colunas.
Checando por número de colunas: .csv tem 30 colunas.
Checando por número de colunas: .csv tem 30 colunas.


### Checando as colunas que não estão presentes em todos os datasets

In [3]:
colunas.columns = ['qtd']
colunas_count = colunas['qtd'].value_counts()
colunas_count = pd.DataFrame(colunas_count)
colunas_count = colunas_count[colunas_count.qtd <8]
colunas_count.index.tolist()

['latitude', 'longitude', 'regional', 'delegacia', 'uop', 'ano']

### Procurando valores nulos no DataFrame total

In [4]:
datatran.isnull().sum()

id                             0
data_inversa                   0
dia_semana                     0
horario                        0
uf                             0
br                           882
km                           882
municipio                      0
causa_acidente                 0
tipo_acidente                 40
classificacao_acidente         1
fase_dia                       1
sentido_via                    0
condicao_metereologica         3
tipo_pista                     0
tracado_via                    0
uso_solo                       0
ano                       515480
pessoas                        0
mortos                         0
feridos_leves                  0
feridos_graves                 0
ilesos                         0
ignorados                      0
feridos                        0
veiculos                       0
latitude                  218524
longitude                 218524
regional                  218530
delegacia                 219412
uop       

Quando verificamos as colunas, já havíamos percebido que as variáveis *ano*, *latitude*, *longitude*, *regional*, *delegacia* e *uop* não estão presentes em todos os datasets. Essas variáveis não serão importantes para a criação de um modelo de previsão, então nos livraremos delas.

Isso nos deixa algumas variáveis que possuem valores nulos e que precisamos decidir acerca do que fazer: a variável *condicao_meterologica*, com 3 valores nulos; a variavel *fase_dia*, com apenas um valor nulo; a variável *classificacao_acidente*, também com apenas um valor nulo; a variável *tipo_acidente*, com 40 valores nulos; a variável *br* e a variável *km*, com 882 valores nulos cada uma.

Começaremos realizando um drop nas colunas que desejamos abandonar.

In [5]:
datatran.drop(columns=colunas_count.index.tolist(),inplace=True)

Podemos igualmente nos livrar das colunas br e km, pois não estamos interessados nas localizações específicas e queremos criar um modelo mais geral.

Como existem poucas observações nulas nas outras colunas em questão, realizaremos a limpeza das mesmas.

In [6]:
datatran.drop(columns=['br','km'],inplace=True)